lonboard/geopandas.py

```diff
@@ -0,0 +1,324 @@
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+
+from . import basemap
+from . import viz
+from .colormap import apply_categorical_cmap, apply_continuous_cmap
+
+__all__ = ["LonboardAccessor"]
+
+
+@pd.api.extensions.register_dataframe_accessor("lb")
+class LonboardAccessor:
+    def __init__(self, pandas_obj):
+        self._validate(pandas_obj)
+        self._obj = pandas_obj
+
+    @staticmethod
+    def _validate(obj):
+        if not isinstance(obj, gpd.GeoDataFrame):
+            raise AttributeError("must be a geodataframe")
+
+    def explore(
+        self,
+        column=None,
+        cmap=None,
+        scheme=None,
+        k=6,
+        categorical=False,
+        elevation=None,
+        extruded=False,
+        elevation_scale=1,
+        alpha=1,
+        layer_kwargs=None,
+        map_kwargs=None,
+        classify_kwargs=None,
+        nan_color=[255, 255, 255, 255],
+        color=None,
+        wireframe=False,
+        tiles="CartoDB Darkmatter",
+        m=None,
+    ):
```

Comment:
Can you expand on what differences (if any) there are with the upstream geopandas.explore method? I think we'd want it to be as similar as possible.

That would be great to include in this docstring.

Additionally, ideally there would be a way to get valid type hinting here, but presumably that won't work if this is added at runtime to the GeoDataFrame instance.

Reply:
sure happy to add both. In general I tried to keep all possible arguments the same (hence the `tiles` that uses a str --> class--> str, which isn't ideal but I'm hard conditioned to type tiles='CartoDB Positron') any time I explore.

the one thing I know is different is `classification_kwargs` used here vs `classification_kwds` used in geopandas, but I'll switch it to martin's version for compat. Otherwise, any arguments available in explore have the same name. Those that *don't* map between the two are unique to the backengs, e.g. `layer_kwargs` here vs `style_kwds` in vanilla explore

also I generally use numpy docstrings but I think the rest of the packages uses another format, so I should probably update that too